components/suggest/src/store.rs

```diff
@@ -708,6 +716,60 @@ mod tests {
 
     use crate::SuggestionProvider;
 
+    /// Builder for `SuggestionQuery` to avoid breaking existing tests when it's
+    /// modified. See also the query macros below.
+    pub(crate) struct SuggestionQueryBuilder {
+        keyword: String,
+        providers: Vec<SuggestionProvider>,
+        limit: Option<i32>,
+        phantom_suggestion_type: Option<String>,
+    }
```

Comment:
I like builders in Rust, but I notice that all of these fields have default values.  We could also use the struct update pattern:
```
SuggestionQuery {
  keyword: "foo",
  providers: vec![...],
  ..SuggestionQuery::default()
}
```

I'm not sure if this is better or worse, I'm happy with whichever one you like.  My one opinion is that I don't think we need to expose this in the UDL, since Kotlin/Swift have similar ways to do the same thing.